pkg/syncer/storagepool/util.go

```diff
@@ -121,40 +121,49 @@ func getHostMoIDToK8sNameMap(ctx context.Context) (map[string]string, error) {
 // clusterID) on which the given datastore is mounted and accessible. The
 // values in the map tells whether the ESX host is in Maintenance Mode.
 func findAccessibleNodes(ctx context.Context, datastore *object.Datastore,
-	clusterID string, vcclient *vim25.Client) (map[string]bool, error) {
+	clusterIDs []string, vcclient *vim25.Client) (map[string]bool, error) {
 	log := logger.GetLogger(ctx)
-	clusterMoref := vimtypes.ManagedObjectReference{
-		Type:  "ClusterComputeResource",
-		Value: clusterID,
-	}
-	cluster := object.NewComputeResource(vcclient, clusterMoref)
-	hosts, err := datastore.AttachedClusterHosts(ctx, cluster)
+
+	nodes := make(map[string]bool)
+	// Now find the k8s node names of these hosts.
+	hostMoIDTok8sName, err := getHostMoIDToK8sNameMap(ctx)
 	if err != nil {
-		log.Infof("Failed to get attached hosts of datastore %s, err=%+v", datastore.Reference().Value, err)
 		return nil, err
 	}
 
-	// Now find the k8s node names of these hosts.
-	hostMoIDTok8sName, err := getHostMoIDToK8sNameMap(ctx)
-	nodes := make(map[string]bool)
-	for _, host := range hosts {
-		thisName, ok := hostMoIDTok8sName[host.Reference().Value]
-		if !ok || thisName == "" {
-			// This host on which the datastore is mounted does not have the
-			// annotation yet. So ignore this node and wait for next reconcile
-			// to pick this up.
-			log.Debugf("Ignoring node %s without vmware-system-esxi-node-moid annotation", host.Reference().Value)
-			continue
+	for _, clusterID := range clusterIDs {
```

Comment:
We don't need to check every cluster. Just check the cluster where this datastore belongs.

Reply:
I've updated the code so that we get the matching cluster ID from `isRemoteVsan` and use that to find accessible nodes.